File: problems/arrays/tournament_winner/js/tournament_winner.js
export const tournamentWinner = (competitions, results) => {
  let currentLeader = { score: 0, name: "" };

  const scores = {};

  for (let i = 0; i < competitions.length; i++) {
    const [homeTeam, awayTeam] = competitions[i];
    const winner = results[i] === 0 ? awayTeam : homeTeam;

    scores[winner] = (scores[winner] || 0) + 3;

    if (scores[winner] > currentLeader.score) {
      currentLeader = { score: scores[winner], name: winner };
    }
  }

  return currentLeader.name;
};

File: problems/arrays/tournament_winner/cpp/tournament_winner.cpp
#include "tournament_winner.h"

std::string tournamentWinner(std::vector<std::vector<std::string>> competitions,
                             std::vector<int> results) {
  std::string winner = "";
  std::map<std::string, int> scores;

  for (auto i = 0; i < results.size(); i++) {
    auto current = competitions[i][!(results[i])];
    scores[current] += 3;
    if (scores[current] > scores[winner])
      winner = current;
  }

  return winner;
}
